Doc updates for My Access - new PR incorporating all feedback from previous reviews (#11361)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/accesspackageassignmentrequest-filterbycurrentuser.md

---

> title: "accessPackageAssignmentRequest: FilterByCurrentUser"

Reviewer: Same point here - likely should be lowercase f.

Author: done

localization_priority: Normal
author: "sbounouh"
ms.prod: "microsoft-identity-platform"
doc_type: "apiPageType"
---

# accessPackageAssignmentRequest: filterByCurrentUser
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

In [Azure AD Entitlement Management](../resources/entitlementmanagement-root.md), retrieve a list of [accessPackageAssignmentRequest](../resources/accesspackageassignmentrequest.md) objects filtered on the signed-in user.

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|EntitlementManagement.Read.All, EntitlementManagement.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported.|
|Application|Not supported.|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
GET /identityGovernance/entitlementManagement/accessPackageAssignmentRequests/filterByCurrentUser
```

> ## Query parameters

Reviewer 2: Should be: Function parameters

> The following table shows the parameters that can be used with this method.

Reviewer 2: ```suggestion
The following table shows the parameters that can be used with the function.
```

|Parameter|Type|Description|
|:---|:---|:---|

> |on|String|The list of options that can be used to filter on current user. Possible options are: `target`, `createdBy` and `approver`. Additional options may be available in the future.|

Reviewer 2: ```suggestion
|on|String|The list of options that can be used to filter on current user. Possible values: `target`, `createdBy` , `approver`.|
```

> - `target` is used to get the `accessPackageAssignmentRequest` objects where the signed-in user is the target. The resulting list includes all the assignment requests, current and expired, that were requested by the called or for the caller, across all catalogs and access packages.

Reviewer 2: ```suggestion
- `target` is used to get the `accessPackageAssignmentRequest` objects where the signed-in user is the target. The resulting list includes all of the assignment requests, current and expired, that were requested by the caller or for the caller, across all catalogs and access packages.
```

> - `createdBy` is used to get the `accessPackageAssignmentRequest` objects created by the signed-in user. The resulting list includes all the assignment requests that the caller has created for themselves or on behalf of others (such as in case of admin direct assignment), across all catalogs and access packages.

Reviewer 2: ```suggestion
- `createdBy` is used to get the `accessPackageAssignmentRequest` objects created by the signed-in user. The resulting list includes all of the assignment requests that the caller has created for themselves or on behalf of others , such as in case of admin direct assignment, across all catalogs and access packages.
```

> - `approver` is used to get the `accessPackageAssignmentRequest` objects where the signed-in user is an allowed approver in any contained `accessPackageAssignment/accessPackageAssignmentPolicy/requestApprovalSettings/approvalStages` (`primaryApprovers` or `escalationApprovers`). The resulting list includes the assignment requests in *pending* state, across all catalogs and access packages and that need a decision from the caller. This list does not contain assignment requests in states other than *pending* nor assignment requests that do not require action from the the signed-in user.

Reviewer 2: ```suggestion
- `approver` is used to get the `accessPackageAssignmentRequest` objects where the signed-in user is an allowed approver in any contained `accessPackageAssignment/accessPackageAssignmentPolicy/requestApprovalSettings/approvalStages` (`primaryApprovers` or `escalationApprovers`). The resulting list includes the assignment requests in *pending* state, across all catalogs and access packages and that need a decision from the caller. This list does not contain assignment requests in states other than `pending`. The list only contains assignment requests that require action from the the signed-in user.
```
